src/k8s/pkg/k8sd/features/calico/status.go

```diff
@@ -0,0 +1,37 @@
+package calico
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/k8s/pkg/snap"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CheckNetwork checks the status of the Calico pods in the Kubernetes cluster.
+// It verifies if all the Calico pods in the "tigera-operator" namespace are ready.
+// If any pod is not ready, it returns false. Otherwise, it returns true.
+func CheckNetwork(ctx context.Context, snap snap.Snap) (bool, error) {
+	client, err := snap.KubernetesClient("calico-system")
+	if err != nil {
+		return false, fmt.Errorf("failed to create kubernetes client: %w", err)
+	}
+
+	calicoPods, err := client.ListPods(ctx, "calico-system", metav1.ListOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to get calico pods: %w", err)
+	}
+
+	for _, pod := range calicoPods {
+		isReady, err := client.IsPodReady(ctx, pod.Name, "calico-system", metav1.ListOptions{})
+		if err != nil {
+			return false, fmt.Errorf("failed to check if pod %q is ready: %w", pod.Name, err)
+		}
+		if !isReady {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
```

Comment:
From your `kubectl get po -A` output, I see that Calico deploys the workloads in three different namespaces:
- `calico-system`
- `calico-apiserver`
- `kube-system`
I believe we should check these three namespaces. This will give us enough confidence that all the components are running correctly.

Reply:
The `tigera-operator` and `calico-apiserver` are somewhat a prerequesite before the calico pods can get ready but you are right - let's be explicit here. I added a check for the other namespace and the operator.